transformation/plugins/org.jboss.tools.fuse.transformation.core/src/org/jboss/tools/fuse/transformation/core/model/ModelBuilder.java

```diff
@@ -64,15 +64,22 @@ private static void addFieldsToModel(List<Field> fields, Model model) {
                 isCollection = true;
                 fieldClass = field.getType().getComponentType();
             } else if (Collection.class.isAssignableFrom(field.getType())) {
-                isCollection = true;
-                Type ft = field.getGenericType();
-                if (ft instanceof ParameterizedType) {
-                    fieldClass = (Class<?>) ((ParameterizedType) ft).getActualTypeArguments()[0];
-                } else {
-                    fieldClass = Object.class;
-                }
+            	isCollection = true;
+            	Type ft = field.getGenericType();
+            	if (ft instanceof ParameterizedType) {
+            		Object testObject = ((ParameterizedType) ft).getActualTypeArguments()[0];
+            		if (testObject instanceof Class) {
+            			fieldClass = (Class<?>) testObject;
+            		} else if (testObject instanceof ParameterizedType) {
```

Comment:
this will avoid the classcast effectively but, is it generating the right thing? What is the right thing to be generated?

Shouldn't be a recursive?